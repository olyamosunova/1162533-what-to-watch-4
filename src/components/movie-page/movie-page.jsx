import React, {PureComponent} from "react";
import PropTypes from "prop-types";
import {connect} from "react-redux";
import MoviesList from "../movies-list/movies-list.jsx";
import MoviePageOverview from "../movie-page-overview/movie-page-overview.jsx";
import MoviePageDetails from "../movie-page-details/movie-page-details.jsx";
import MoviePageReviews from "../movie-page-reviews/movie-page-reviews.jsx";
import Header from "../header/header.jsx";
import {TabsName} from "../../const";
import {ActionCreator} from "../../reducer/states/states";
import {getMovies, getReviews} from "../../reducer/data/selectors";
import {getAuthorizationStatus} from "../../reducer/user/selectors";
import {AuthorizationStatus} from "../../const";
import MyListButton from "../my-list-button/my-list-button.jsx";
import {Operations} from "../../reducer/data/data";
import {getCurrentPage} from "../../reducer/states/selectors";
import {Link} from "react-router-dom";
import {AppRoute} from "../../const";

const SIMILAR_FILM_COUNT = 4;

class MoviePage extends PureComponent {
  constructor(props) {
    super(props);
  }

  componentDidMount() {
    const {movie, loadMovie} = this.props;
    loadMovie(movie);
  }

  componentDidUpdate() {
    const {movie, loadMovie} = this.props;
    loadMovie(movie);
  }


  render() {
    const {
      movie,
      movies,
      renderTabs,
      activeTab,
      onPlayClick,
      reviews,
      isSignedIn,
      onMyListClick,
      currentPage,
    } = this.props;

    const {promoMovie, backgroundColor, isFavorite} = movie;
    const {id, title, genre, releaseDate, poster, cover} = promoMovie;

    const addReviewButton = (
      <Link
        to={`${AppRoute.MOVIE}/${id}/review`}
        className="btn movie-card__button">
        Add review
      </Link>
    );

    const getSimilarMovies = (currentGenre, films, currentId) => {
      const similarMovies = films.filter((film) => {
        if (currentId !== film.promoMovie.id && film.promoMovie.genre === currentGenre) {
          return film;
        }
        return null;
      }).slice(0, SIMILAR_FILM_COUNT);
      return similarMovies;
    };


    const _renderTabsInformation = () => {
      switch (activeTab) {
        case TabsName.OVERVIEW:
          return <MoviePageOverview
            movie={movie}
          />;
        case TabsName.DETAILS:
          return <MoviePageDetails
            movie={movie}
          />;
        case TabsName.REVIEWS:
          return <MoviePageReviews reviews={reviews} />;
      }

      return null;
    };

    return (
      <React.Fragment>
        <section className="movie-card movie-card--full" style={{backgroundColor: `${backgroundColor}`}}>
          <div className="movie-card__hero">
            <div className="movie-card__bg">
              <img src={cover} alt={title}/>
            </div>

            <h1 className="visually-hidden">WTW</h1>

            <Header currentPage={currentPage} />

            <div className="movie-card__wrap">
              <div className="movie-card__desc">
                <h2 className="movie-card__title">{title}</h2>
                <p className="movie-card__meta">
                  <span className="movie-card__genre">{genre}</span>
                  <span className="movie-card__year">{releaseDate}</span>
                </p>

                <div className="movie-card__buttons">
                  <button
                    className="btn btn--play movie-card__button"
                    type="button"
                    onClick={() => {
                      onPlayClick(movie);
                    }}
                  >
                    <svg viewBox="0 0 19 19" width="19" height="19">
                      <use xlinkHref="#play-s"></use>
                    </svg>
                    <span>Play</span>
                  </button>
                  <MyListButton id={id} isFavorite={isFavorite} onMyListClick={onMyListClick} />
                  {isSignedIn && addReviewButton}
                </div>
              </div>
            </div>
          </div>

          <div className="movie-card__wrap movie-card__translate-top">
            <div className="movie-card__info">
              <div className="movie-card__poster movie-card__poster--big">
                <img
                  src={poster}
                  alt="The Grand Budapest Hotel poster"
                  width="218"
                  height="327"/>
              </div>

              <div className="movie-card__desc">
                {renderTabs()}
                {_renderTabsInformation()}
              </div>
            </div>
          </div>
        </section>

        <div className="page-content">
          <section className="catalog catalog--like-this">
            <h2 className="catalog__title">More like this</h2>

            <MoviesList
              movies={getSimilarMovies(genre, movies, movie.promoMovie.id)}
              genre={genre}
            />
          </section>

          <footer className="page-footer">
            <div className="logo">
              <a href="main.html" className="logo__link logo__link--light">
                <span className="logo__letter logo__letter--1">W</span>
                <span className="logo__letter logo__letter--2">T</span>
                <span className="logo__letter logo__letter--3">W</span>
              </a>
            </div>

            <div className="copyright">
              <p>Â© 2019 What to watch Ltd.</p>
            </div>
          </footer>
        </div>
      </React.Fragment>
    );
  }
}

MoviePage.propTypes = {
  movie: PropTypes.shape({
    promoMovie: PropTypes.shape({
      id: PropTypes.number.isRequired,
      title: PropTypes.string.isRequired,
      genre: PropTypes.string.isRequired,
      releaseDate: PropTypes.number.isRequired,
      poster: PropTypes.string.isRequired,
      cover: PropTypes.string.isRequired,
      previewVideo: PropTypes.string.isRequired,
    }),
    backgroundColor: PropTypes.string.isRequired,
    rating: PropTypes.number.isRequired,
    ratingLevel: PropTypes.string.isRequired,
    ratingCount: PropTypes.number.isRequired,
    runTime: PropTypes.number.isRequired,
    description: PropTypes.string.isRequired,
    director: PropTypes.string.isRequired,
    starring: PropTypes.array.isRequired,
    isFavorite: PropTypes.bool.isRequired,
  }),
  movies: PropTypes.arrayOf(
      PropTypes.shape({
        promoMovie: PropTypes.shape({
          id: PropTypes.number.isRequired,
          title: PropTypes.string.isRequired,
          genre: PropTypes.string.isRequired,
          releaseDate: PropTypes.number.isRequired,
          poster: PropTypes.string.isRequired,
          cover: PropTypes.string.isRequired,
          previewVideo: PropTypes.string.isRequired,
        }),
        backgroundColor: PropTypes.string.isRequired,
        rating: PropTypes.number.isRequired,
        ratingLevel: PropTypes.string.isRequired,
        ratingCount: PropTypes.number.isRequired,
        runTime: PropTypes.number.isRequired,
        description: PropTypes.string.isRequired,
        director: PropTypes.string.isRequired,
        starring: PropTypes.array.isRequired,
        isFavorite: PropTypes.bool.isRequired,
      })),
  reviews: PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.number.isRequired,
        message: PropTypes.string.isRequired,
        rating: PropTypes.number.isRequired,
        author: PropTypes.string.isRequired,
        date: PropTypes.string.isRequired,
      })),
  renderTabs: PropTypes.func.isRequired,
  activeTab: PropTypes.string.isRequired,
  onPlayClick: PropTypes.func.isRequired,
  isSignedIn: PropTypes.bool.isRequired,
  onMyListClick: PropTypes.func.isRequired,
  currentPage: PropTypes.string.isRequired,
  loadMovie: PropTypes.func.isRequired,
};

const mapStateToProps = (state) => {
  return {
    movies: getMovies(state),
    reviews: getReviews(state),
    isSignedIn: getAuthorizationStatus(state) === AuthorizationStatus.AUTH,
    currentPage: getCurrentPage(state),
  };
};

const mapDispatchToProps = (dispatch) => ({
  onPlayClick(movie) {
    dispatch(ActionCreator.chooseMovieToWatch(movie));
  },
  onMyListClick(movieId, status, isPromoMovie) {
    dispatch(Operations.changeFlagIsFavorite(movieId, status, isPromoMovie));
  },
  loadMovie(movie) {
    dispatch(Operations.loadReviews(movie.promoMovie.id));
  }
});

export {MoviePage};
export default connect(mapStateToProps, mapDispatchToProps)(MoviePage);
