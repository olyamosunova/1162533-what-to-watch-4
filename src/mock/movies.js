const PromoMovie = {
  TITLE: `The Grand Budapest Hotel`,
  GENRE: `Drama`,
  RELEASE_DATE: 2014,
  POSTER: `the-grand-budapest-hotel-poster.jpg`,
  COVER: `bg-the-grand-budapest-hotel.jpg`,
};

const MoviesList = [
  {
    id: 1,
    title: `Fantastic Beasts: The Crimes of Grindelwald`,
    poster: `fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  },
  {
    id: 2,
    title: `Bohemian Rhapsody`,
    poster: `bohemian-rhapsody.jpg`,
  },
  {
    id: 3,
    title: `Macbeth`,
    poster: `macbeth.jpg`,
  },
  {
    id: 4,
    title: `Aviator`,
    poster: `aviator.jpg`,
  },
  {
    id: 5,
    title: `We need to talk about Kevin`,
    poster: `we-need-to-talk-about-kevin.jpg`,
  },
  {
    id: 6,
    title: `What We Do in the Shadows`,
    poster: `what-we-do-in-the-shadows.jpg`,
  },
  {
    id: 7,
    title: `Revenant`,
    poster: `revenant.jpg`,
  },
  {
    id: 8,
    title: `Johnny English`,
    poster: `johnny-english.jpg`,
  },
];

const MovieDescriptions = {
  TITLE: `The Grand Budapest Hotel`,
  GENRE: `Drama`,
  RELEASE_DATE: 2014,
  POSTER: `the-grand-budapest-hotel-poster.jpg`,
  COVER: `bg-the-grand-budapest-hotel.jpg`,
  RATING: 8.9,
  RATING_LEVEL: `Very good`,
  RATING_COUNT: 240,
  DESCRIPTION: [
    `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
    `Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
  ],
  DIRECTOR: `Wes Andreson`,
  STARRING: `Bill Murray, Edward Norton, Jude Law, Willem Dafoe and other`,
};

export {PromoMovie, MoviesList, MovieDescriptions};
