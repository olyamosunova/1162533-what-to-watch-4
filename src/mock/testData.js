const genres = [`Drama`, `Comedy`, `Western`];

const movie = {
  promoMovie: {
    id: 1,
    title: `Macbeth`,
    poster: `macbeth.jpg`,
    cover: `macbeth.jpg`,
    genre: `Drama`,
    releaseDate: 2015,
    previewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
  backgroundColor: ``,
  previewImage: ``,
  videoLink: ``,
  rating: 7.1,
  ratingLevel: `Very good`,
  ratingCount: 240,
  runTime: 120,
  description: ``,
  director: `Wes Andreson`,
  starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Willem Dafoe`, `Saoirse Ronan`,
    `Tony Revoloru`, `Tilda Swinton`, `Tom Wilkinson`],
};

const Movies = [
  {
    promoMovie: {
      id: 1,
      title: `Fantastic Beasts: The Crimes of Grindelwald`,
      poster: `fantastic-beasts-the-crimes-of-grindelwald.jpg`,
      cover: `fantastic-beasts-the-crimes-of-grindelwald.jpg`,
      genre: `Fantasy`,
      releaseDate: 2018,
      previewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    },
    backgroundColor: ``,
    previewImage: ``,
    videoLink: ``,
    rating: 6.6,
    ratingLevel: `Very good`,
    ratingCount: 240,
    runTime: 110,
    description: ``,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Willem Dafoe`, `Saoirse Ronan`,
      `Tony Revoloru`, `Tilda Swinton`, `Tom Wilkinson`],
  },
  {
    promoMovie: {
      id: 2,
      title: `Bohemian Rhapsody`,
      poster: `bohemian-rhapsody.jpg`,
      cover: `bohemian-rhapsody.jpg`,
      genre: `Music`,
      releaseDate: 2018,
      previewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    },
    backgroundColor: ``,
    previewImage: ``,
    videoLink: ``,
    rating: 8.0,
    ratingLevel: `Very good`,
    ratingCount: 240,
    runTime: 110,
    description: ``,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Willem Dafoe`, `Saoirse Ronan`,
      `Tony Revoloru`, `Tilda Swinton`, `Tom Wilkinson`],
  },
  {
    promoMovie: {
      id: 3,
      title: `Macbeth`,
      poster: `macbeth.jpg`,
      cover: `macbeth.jpg`,
      genre: `Drama`,
      releaseDate: 2015,
      previewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    },
    backgroundColor: ``,
    previewImage: ``,
    videoLink: ``,
    rating: 7.1,
    ratingLevel: `Very good`,
    ratingCount: 240,
    runTime: 110,
    description: ``,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Willem Dafoe`, `Saoirse Ronan`,
      `Tony Revoloru`, `Tilda Swinton`, `Tom Wilkinson`],
  },
  {
    promoMovie: {
      id: 4,
      title: `Aviator`,
      poster: `aviator.jpg`,
      cover: `aviator.jpg`,
      genre: `Drama`,
      releaseDate: 2014,
      previewVideo: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    },
    backgroundColor: ``,
    previewImage: ``,
    videoLink: ``,
    rating: 7.6,
    ratingLevel: `Very good`,
    ratingCount: 240,
    runTime: 110,
    description: ``,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Willem Dafoe`, `Saoirse Ronan`,
      `Tony Revoloru`, `Tilda Swinton`, `Tom Wilkinson`],
  },
  {
    promoMovie: {
      id: 5,
      title: `We need to talk about Kevin`,
      poster: `we-need-to-talk-about-kevin.jpg`,
      cover: `we-need-to-talk-about-kevin.jpg`,
      genre: `Drama`,
      releaseDate: 2011,
      previewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    },
    backgroundColor: ``,
    previewImage: ``,
    videoLink: ``,
    rating: 7.5,
    ratingLevel: `Very good`,
    ratingCount: 240,
    runTime: 110,
    description: ``,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Willem Dafoe`, `Saoirse Ronan`,
      `Tony Revoloru`, `Tilda Swinton`, `Tom Wilkinson`],
  },
  {
    promoMovie: {
      id: 6,
      title: `What We Do in the Shadows`,
      poster: `what-we-do-in-the-shadows.jpg`,
      cover: `what-we-do-in-the-shadows.jpg`,
      genre: `Comedy`,
      releaseDate: 2014,
      previewVideo: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    },
    backgroundColor: ``,
    previewImage: ``,
    videoLink: ``,
    rating: 7.4,
    ratingLevel: `Very good`,
    ratingCount: 240,
    runTime: 110,
    description: ``,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Willem Dafoe`, `Saoirse Ronan`,
      `Tony Revoloru`, `Tilda Swinton`, `Tom Wilkinson`],
  },
  {
    promoMovie: {
      id: 7,
      title: `Revenant`,
      poster: `revenant.jpg`,
      cover: `revenant.jpg`,
      genre: `Vestern`,
      releaseDate: 2015,
      previewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,

    },
    backgroundColor: ``,
    previewImage: ``,
    videoLink: ``,
    rating: 7.8,
    ratingLevel: `Very good`,
    ratingCount: 240,
    runTime: 110,
    description: ``,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Willem Dafoe`, `Saoirse Ronan`,
      `Tony Revoloru`, `Tilda Swinton`, `Tom Wilkinson`],
  },
  {
    promoMovie: {
      id: 8,
      title: `Johnny English`,
      poster: `johnny-english.jpg`,
      cover: `johnny-english.jpg`,
      genre: `Comedy`,
      releaseDate: 2003,
      previewVideo: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    },
    backgroundColor: ``,
    previewImage: ``,
    videoLink: ``,
    rating: 6.2,
    ratingLevel: `Very good`,
    ratingCount: 240,
    runTime: 110,
    description: ``,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Willem Dafoe`, `Saoirse Ronan`,
      `Tony Revoloru`, `Tilda Swinton`, `Tom Wilkinson`],
  },
  {
    promoMovie: {
      id: 9,
      title: `Fantastic Beasts: The Crimes of Grindelwald`,
      poster: `fantastic-beasts-the-crimes-of-grindelwald.jpg`,
      cover: `fantastic-beasts-the-crimes-of-grindelwald.jpg`,
      genre: `Fantasy`,
      releaseDate: 2018,
      previewVideo: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    },
    backgroundColor: ``,
    previewImage: ``,
    videoLink: ``,
    rating: 6.6,
    ratingLevel: `Very good`,
    ratingCount: 240,
    runTime: 110,
    description: ``,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Willem Dafoe`, `Saoirse Ronan`,
      `Tony Revoloru`, `Tilda Swinton`, `Tom Wilkinson`],
  },
  {
    promoMovie: {
      id: 10,
      title: `Bohemian Rhapsody`,
      poster: `bohemian-rhapsody.jpg`,
      cover: `bohemian-rhapsody.jpg`,
      genre: `Music`,
      releaseDate: 2018,
      previewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    },
    backgroundColor: ``,
    previewImage: ``,
    videoLink: ``,
    rating: 8.0,
    ratingLevel: `Very good`,
    ratingCount: 240,
    runTime: 110,
    description: ``,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Willem Dafoe`, `Saoirse Ronan`,
      `Tony Revoloru`, `Tilda Swinton`, `Tom Wilkinson`],
  },
];

const reviews = [{
  id: 1,
  message: ``,
  rating: 8.9,
  author: ``,
  date: ``,
}];

const userData = {
  id: 0,
  email: `vesan@mail.ru`,
  name: `Lola`,
  avatarUrl: ``,
};

export {movie, Movies, reviews, genres, userData};
