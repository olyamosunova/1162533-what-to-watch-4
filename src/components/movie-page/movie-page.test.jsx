import React from "react";
import renderer from "react-test-renderer";
import {MoviePage} from "./movie-page.jsx";
import {movie, Movies, userData} from "../../mock/testData";
import {Router} from 'react-router-dom';
import history from '../../history';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import NameSpace from "../../reducer/name-space";

const mockStore = configureStore([]);

const store = mockStore({
  [NameSpace.DATA]: {
    movies: Movies,
  },
  [NameSpace.USER]: {
    authorizationStatus: `AUTH`,
    userData: {
      id: 1,
      email: `sadas@gmail.com`,
      name: `asdasd`,
      avatarUrl: `https://4.react.pages.academy/wtw/asda.jpg`,
    }
  },
});

it(`Should MoviePage render correctly`, () => {
  const tree = renderer
    .create(
        <Router history={history}>
          <Provider store={store}>
            <MoviePage
              movie={movie}
              movies={Movies}
              onMovieClick={() => {}}
              renderTabs={() => {}}
              activeTab={`Overview`}
              authorizationStatus={`NO_AUTH`}
              userData={userData}
              onLoginClick={() => {}}
              isSignedIn={true}
              onAddReviewClick={() => {}}
              onMyListClick={() => {}}
              loadMovie={() => {}}
            />
          </Provider>
        </Router>, {
          createNodeMock: ()=>{
            return {};
          }
        })
    .toJSON();

  expect(tree).toMatchSnapshot();
});
