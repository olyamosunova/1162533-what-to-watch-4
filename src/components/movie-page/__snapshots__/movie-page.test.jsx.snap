// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should MoviePage render correctly 1`] = `
Array [
  <section
    className="movie-card movie-card--full"
    style={
      Object {
        "backgroundColor": "",
      }
    }
  >
    <div
      className="movie-card__hero"
    >
      <div
        className="movie-card__bg"
      >
        <img
          alt="Macbeth"
          src="macbeth.jpg"
        />
      </div>
      <h1
        className="visually-hidden"
      >
        WTW
      </h1>
      <header
        className="page-header movie-card__head"
      >
        <div
          className="logo"
        >
          <a
            className="logo__link"
            href="main.html"
          >
            <span
              className="logo__letter logo__letter--1"
            >
              W
            </span>
            <span
              className="logo__letter logo__letter--2"
            >
              T
            </span>
            <span
              className="logo__letter logo__letter--3"
            >
              W
            </span>
          </a>
        </div>
        <div
          className="user-block"
        >
          <a
            className="user-block__link"
            href="sign-in.html"
            onClick={[Function]}
          >
            Sign in
          </a>
        </div>
      </header>
      <div
        className="movie-card__wrap"
      >
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            Macbeth
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              Drama
            </span>
            <span
              className="movie-card__year"
            >
              2015
            </span>
          </p>
          <div
            className="movie-card__buttons"
          >
            <button
              className="btn btn--play movie-card__button"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlinkHref="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </button>
            <button
              className="btn btn--list movie-card__button"
              type="button"
            >
              <svg
                height="20"
                viewBox="0 0 19 20"
                width="19"
              >
                <use
                  xlinkHref="#add"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
            <a
              className="btn movie-card__button"
              href="add-review.html"
              onClick={[Function]}
            >
              Add review
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      className="movie-card__wrap movie-card__translate-top"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster movie-card__poster--big"
        >
          <img
            alt="The Grand Budapest Hotel poster"
            height="327"
            src="macbeth.jpg"
            width="218"
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <div
            className="movie-rating"
          >
            <div
              className="movie-rating__score"
            >
              7.1
            </div>
            <p
              className="movie-rating__meta"
            >
              <span
                className="movie-rating__level"
              >
                Very good
              </span>
              <span
                className="movie-rating__count"
              >
                240
                 ratings
              </span>
            </p>
          </div>
          <div
            className="movie-card__text"
          >
            
            <p
              className="movie-card__director"
            >
              <strong>
                Director:
                Wes Andreson
              </strong>
            </p>
            <p
              className="movie-card__starring"
            >
              <strong>
                Starring:
                Bill Murray,Edward Norton,Jude Law,Willem Dafoeand other
              </strong>
            </p>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <div
    className="page-content"
  >
    <section
      className="catalog catalog--like-this"
    >
      <h2
        className="catalog__title"
      >
        More like this
      </h2>
      <div
        className="catalog__movies-list"
      >
        <article
          className="small-movie-card catalog__movies-card"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              height="100%"
              poster=""
              width="100%"
            >
              <source
                src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
              />
            </video>
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
              onClick={[Function]}
            >
              Macbeth
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              height="100%"
              poster=""
              width="100%"
            >
              <source
                src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
              />
            </video>
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
              onClick={[Function]}
            >
              Aviator
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              height="100%"
              poster=""
              width="100%"
            >
              <source
                src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
              />
            </video>
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
              onClick={[Function]}
            >
              We need to talk about Kevin
            </a>
          </h3>
        </article>
      </div>
    </section>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
          href="main.html"
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          Â© 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
]
`;
